src/pages/Locations/locations.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { useState, useEffect } from 'react'
+import {API} from "../../services/api";
+import {DataGrid, GridColumns} from '@mui/x-data-grid'
+
+export const LocationGrid = () => {
+    const [tableData, setTableData] = useState([])
+    const columns: GridColumns = [
+        { field: 'name',  headerName: 'Name', width: 150, },
+        { field: 'street', headerName: 'Street', width: 220 },
+        { field: 'town', headerName: 'Town', width: 150 },
+        { field: 'zip', headerName: 'Zip' , width: 150},
+        // { field: 'owner', headerName: 'Owner', width: 200}
+    ]
+
+    useEffect(() => {
+            API.getLocations().then((data)=> JSON.stringify(data)).then((data)=>JSON.parse(data)).then((data)=>setTableData(data))},[])
+    return (
+        <div style={{ height: 600, width: '66%' }}>
+            <DataGrid
```

Comment:
here we can add a loader while the Api is getting the data:

`tableData != null ? <DataGrid/> : <LoaderComponent/>`